lib/agent/redirect.js

```diff
@@ -83,16 +56,38 @@ class RedirectHandler {
 
   onData (chunk) {
     if (this.location) {
-      // Ignore body.
-      // TODO (fix): Spec ref?
+      /*
+        https://tools.ietf.org/html/rfc7231#section-6.4
+
+        TLDR: undici always ignores 3xx response bodies.
+
+        Redirection is used to serve the requested resource from another URL, so it is assumes that
+        no body is generated (and thus can be ignored). Even thought generating a body is not prohibited.
+
+        For status 301, 302, 303, 307 and 308 (the latter from RFC 7238), the specs mention that the body usually
+        (which means it's optional and not mandated) contain just an hyperlink to the value of
+        the Location response header, so the body can be ignored safely.
+
+        For status 300, which is "Multiple Choices", the spec mentions both generating a Location
+        response header AND a response body with the other possible location to follow.
+        Since the spec explicitily chooses not to specify a format for such body and leave it to
+        servers and browsers implementors, we ignore the body as there is no specified way to eventually parse it.
+      */
     } else {
       return this.handler.onData(chunk)
     }
   }
 
   onComplete (trailers) {
     if (this.location) {
-      assert(!trailers || trailers.length === 0, 'redirect should not have trailers')
```

Comment:
This can happen. Please put back.

Reply:
I'm not sure I understand.
In the comment I'm not saying it cannot happen, I'm saying that we don't handle them, like we do with the body.

I just forgot to remove the `this.agent.dispatch` line below. Do you agree or do you explicitily want to handle trailers?

Reply:
I would add a comment so that it's explicit that we ignore them on purpose

Reply:
Ok, and then I'll redirect to the full explanation above. Understood. Updated coming soon.

Reply:
Done!